### ISSUE TITLE
I cannot reproduced the results follow the giving training and evaluation files

### ISSUE THREAD

**Issue by leoozy** [princeton-nlp/SimCSE#25]
Hello, thank you for your released code. I run the "run_unsup_example.sh" without any modification. After one epoch of training, I got the model. Then I evaluated the model by :

`python evaluation.py \
    --model_name_or_path PATH_TO_MY_MODEL_FOLDER \
    --pooler cls \
    --task_set sts \
    --mode test`
I got the following results
![image](https://user-images.githubusercontent.com/26025961/117770348-64f3b180-b267-11eb-8bcb-4b6bf092f103.png)
 Furthermore, I tires the uploaded pretrained models. The accuracy is correct. Is there something wrong during my traing?
My envronment:
Pytorch-1.8.0
CUDA-11.1
NVIDIA V100

**Comment by yaoxingcheng** [comment]
During training we have an extra MLP layer on top of BERT encoder. I think you should use `simcse_to_huggingface.py` first to convert the trained model to standard huggingface setting, then run the evaluation.

**Comment by Doragd** [comment]
> Thank you for your reply, I will try it!

My Environment:
Pytorch-1.8.0
CUDA-11.1
RTX 8000

Could you report your results?
I got the following results:
![image](https://user-images.githubusercontent.com/26213546/117831288-ff271a00-b2a6-11eb-8399-903823f597a2.png)

**Comment by leoozy** [comment]
> > Thank you for your reply, I will try it!
> 
> My Environment:
> Pytorch-1.8.0
> CUDA-11.1
> RTX 8000
> 
> Could you report your results?
> I got the following results:
> ![image](https://user-images.githubusercontent.com/26213546/117831288-ff271a00-b2a6-11eb-8399-903823f597a2.png)
> @leoozy
Our results are similar.
![image](https://user-images.githubusercontent.com/26025961/117919085-f7a35780-b31e-11eb-8067-dabf98892ee3.png)

**Comment by gaotianyu1350** [comment]
Hi guys,

We just found a new set of hyperparameters that works better

|               | Unsup. BERT | Unsup. RoBERTa | Sup.      |
|:--------------|:-----------:|:--------------:|:---------:|
| Batch size    | 64          | 512            | 512       |
| Learning rate (base)  | 3e-5 | 1e-5 | 5e-5 |
| Learning rate (large) | 1e-5 | 3e-5 | 1e-5 |

We have also found that for unsupervised models, discarding the MLP layer over CLS works better. We have already updated our training code and example script. During test, you should use `--pooler cls_before_pooler` option. 

For your reference, now unsupervised SimCSE-BERT-base can achieve average STS of `76.25`. Let me know if you encounter any problem when reproducing the results.

**Comment by Doragd** [comment]
> Hi guys,
> 
> We just found a new set of hyperparameters that works better
> 
> Unsup. BERT	Unsup. RoBERTa	Sup.
> Batch size	64	512	512
> Learning rate (base)	3e-5	1e-5	5e-5
> Learning rate (large)	1e-5	3e-5	1e-5
> We have also found that for unsupervised models, discarding the MLP layer over CLS works better. We have already updated our training code and example script. During test, you should use `--pooler cls_before_pooler` option.
> 
> For your reference, now unsupervised SimCSE-BERT-base can achieve average STS of `76.25`. Let me know if you encounter any problem when reproducing the results.

thanks, I got a new average STS of `75.24`.
![image](https://user-images.githubusercontent.com/26213546/118101489-424dce00-b40a-11eb-8129-536a71444ff7.png)
